hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs

```diff
@@ -53,7 +54,8 @@ import           Haskell.Ide.Engine.PluginsIdeMonads
 modifyCache :: (HasGhcModuleCache m) => (GhcModuleCache -> GhcModuleCache) -> m ()
 modifyCache f = do
   mc <- getModuleCache
-  setModuleCache (f mc)
+  let !cached = f mc -- Avoid a space leak by forcing the cache calculation
+  setModuleCache cached
```

Comment:
Should we not make `setModuleCache` itself strict in its argument?  Otherwise we have to do this in all places it might get used.

Reply:
Sounds good, I'll do that instead